src/DiagPDF.cpp

```diff
@@ -0,0 +1,337 @@
+#include <ios>
+
+#include "AMReX_BLassert.H"
+#include "AMReX_Extension.H"
+#include "AMReX_GpuContainers.H"
+#include "AMReX_MultiFabUtil.H"
+#include "AMReX_ParmParse.H"
+#include "AMReX_SPACE.H"
+
+#include "DiagPDF.H"
+
+void DiagPDF::init(const std::string &a_prefix, std::string_view a_diagName)
+{
+	DiagBase::init(a_prefix, a_diagName);
+
+	amrex::ParmParse const hist_pp(a_prefix);
+	hist_pp.query("normalized", m_normalized);
+	hist_pp.query("volume_weighted", m_volWeighted);
```

Comment:
What is the alternative to volume weighted? It seems like it is cell count-weighted, but it is not obvious to me that this is a useful thing to offer or be able to compute. Are there any real circumstances where you'd want cell count-weighted rather than volume-weighted? It seems like the obvious alternative is mass-weighted, and if you don't specify that I would naively expect that selecting volume-weighted to be false would result in mass-weighted, not cell-count weighted. I would suggest either making this the behavior, or, if you want to preserve cell count weighting as an option, changing this to an integer value that can be 0, 1, or 2, with each numerical value corresponding to one of the alternatives for weighting (cell count, volume, mass).

Reply:
That's a good point. I'll change it so it can be weighted one of  (cell count, volume, mass) with volume as the default.